src/approaches/grammar_search_invention_approach.py

```diff
@@ -54,12 +54,18 @@ def _create_grammar(grammar_name: str, dataset: Dataset,
         # For each predicate enumerated, we also enumerate foralls for
         # that predicate, along with appropriate negations.
         forall_grammar = _ForallPredicateGrammarWrapper(negated_grammar)
-        # Finally, we don't actually need to enumerate the given predicates
+        # Prune proposed predicates by checking if they are equivalent to
```

Comment:
it's a bit weird to do this for `forall_single_feat_ineqs` but not `single_feat_ineqs`. since these names are extremely out of date, can we get rid of the latter and just have the former (with a better name)? In fact, I don't think we need holding_dummy anymore, it's served its purpose. Should we just get rid of `CFG.grammar_search_grammar_name` altogether?